lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,394 @@
+DEFAULT_HEADER["TELESCOP"] = "CTA-N"
+DEFAULT_HEADER["CREATED"] = Time.now().utc.iso
+
+location = EarthLocation.from_geodetic(-17.89139 * u.deg, 28.76139 * u.deg, 2184 * u.m)
+
+wobble_offset = 0.4
+
+
+def create_obs_hdu_index(
+    filename_list,
+    fits_dir,
+    hdu_index_filename,
+    obs_index_filename,
+    add_fits_dir_meta=False,
+):
+    """
+    Create the obs table and hdu table
+
+    A two-level index file scheme is used in the IACT community to allow
+    an arbitrary folder structures. For each directory tree, two files should be present:
+    1. obs-index.fits.gz
+    (http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/obs_index/index.html)
+    2. hdu-index.fits.gz
+    (http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/hdu_index/index.html)
+
+    obs-index contains the single run informations e.g.
+    (OBS_ID, RA_PNT, DEC_PNT, ZEN_PNT, ALT_PNT)
+    while hdu-index contains the informations about the locations of the other
+    HDU (header data units) necessary to the analysis.
+
+    File directory information is currently kept optional in the HDU index table
+
+    Parameters
+    ----------
+    filename_list : list
+        list of filenames
+    fits_dir : Path
+        directory containing the fits file
+    hdu_index_filename : str
+        filename for HDU index file
+    obs_index_filename : str
+        filename for OBS index file
+    add_fits_dir_meta : Bool
+        True for adding directory path of FITS file in HDU index table
+
+    Returns
+    -------
+    hdu_index_list : 'astropy.io.fits.HDUList'
+        HDU list file for HDU index tables
+    obs_index_list : 'astropy.io.fits.HDUList'
+        HDU list file for Obs index table
+    """
+
+    hdu_index_tables = []
+    obs_index_tables = []
+
+    # loop through the files
+    for file in filename_list:
+        filepath = fits_dir / file
+        if filepath.is_file():
+            try:
+                event_table = Table.read(filepath, hdu="EVENTS")
+            except Exception:
+                log.error(f"fits corrupted for file {file}")
+                continue
+        else:
+            log.error(f"fits {file} doesn't exist")
+            continue
+
+        # The column names for the table follows the scheme as shown in
+        # https://gamma-astro-data-formats.readthedocs.io/en/latest/general/HDU_CLASS.html
+        # Event list
+        t_events = {
+            "OBS_ID": event_table.meta["OBS_ID"],
+            "HDU_TYPE": "events",
+            "HDU_CLASS": "events",
+            "HDU_CLASS1": "EVENTS",
+            "HDU_CLASS2": "",
+            "HDU_CLASS3": "",
+            "HDU_CLASS4": "",
+            "FILE_DIR": "",
+            "FILE_NAME": file,
+            "HDU_NAME": "EVENTS",
+        }
+        if add_fits_dir_meta:
+            t_events["FILE_DIR"] = str(fits_dir)
+
+        hdu_index_tables.append(t_events)
+
+        # GTI
+        t_gti = t_events.copy()
```

Comment:
You don't actually check that there is a `GTI` table in the files, but add it to the HDU index anyway